packages/hoprd/src/api/v2/paths/withdraw.ts

```diff
@@ -0,0 +1,114 @@
+import type { Operation } from 'express-openapi'
+import type Hopr from '@hoprnet/hopr-core'
+import { Address } from '@hoprnet/hopr-utils'
+import BN from 'bn.js'
+import { STATUS_CODES } from '../'
+
+export const withdraw = async (node: Hopr, rawCurrency: 'NATIVE' | 'HOPR', recipient: string, amount: string) => {
+  const currency = rawCurrency.toUpperCase() as 'NATIVE' | 'HOPR'
+  if (!['NATIVE', 'HOPR'].includes(currency)) {
+    throw Error(STATUS_CODES.INVALID_CURRENCY)
+  }
+
+  if (isNaN(Number(amount))) {
+    throw Error(STATUS_CODES.INVALID_AMOUNT)
+  }
+
+  try {
+    Address.fromString(recipient)
+  } catch (_err) {
+    throw Error(STATUS_CODES.INVALID_ADDRESS)
+  }
+
+  const balance = currency === 'NATIVE' ? await node.getNativeBalance() : await node.getBalance()
+  if (balance.toBN().lt(new BN(amount))) {
+    throw Error(STATUS_CODES.NOT_ENOUGH_BALANCE)
+  }
+
+  const txHash = await node.withdraw(currency, recipient, amount)
+  return txHash
+}
+
+export const POST: Operation = [
```

Comment:
withdrawing HOPR seems to be stuck loading

Reply:
On my mashine it takes ~20s to withdraw hopr both with api and cli, don't know exacly whats the normal time it should take. Commands code for withdraw was not touched so can you try using it and see if there is any difference in wait time?